compiler-rt/test/hwasan/TestCases/hwasan_symbolize_stack_overflow.cpp

```diff
@@ -0,0 +1,29 @@
+// RUN: %clang_hwasan -Wl,--build-id -g %s -o %t
```

Comment:
Why not a more typical pattern, and avoid all the redundant calls to `dirname`?
```suggestion
// RUN: rm -rf %t && mkdir %t
// RUN: %clang_hwasan -Wl,--build-id -g %s -o %t/overflow
```
When you reland, you may want to try that pattern instead.

Reply:
But I actually need the binary to be in the directory, so I don't think this is going to work.

Reply:
I'm not seeing why it wouldn't work w/ `%t` as the directory? I'd think that `hwasan_symbolize --symbols %t --index` would work just as well, right? same w/ the `%run` bits.

There's also always `cd` if you need things in the CWD. Downside is that you can't just copy paste commands from lit anymore w/o using `cd`  yourself, but I know LLD tests use this pattern frequently.

While its deprecated there is also `%T`. That has issues w/ races though, so even then I'm not terribly sure you want to use it, but you don't need `$(dirname %t)` calls throughout.